Frontend/student-hub/src/pages/Friends.js

```diff
@@ -0,0 +1,279 @@
+        const token = sessionStorage.getItem('token');
+        const response = await fetch('http://localhost:8080/friends/remove', {
+            method: 'DELETE',
+            headers: {
+                'Content-Type': 'application/json',
+                Authorization: `Bearer ${token}`
+            },
+            body: JSON.stringify({ userId, friendId })
+        });
+
+        if (!response.ok) {
+            throw new Error(`Failed to remove friend: ${response.statusText}`);
+        }
+        //After successfully removing friend, fetch updated data
+        await fetchFriends();
+        await fetchAvailableFriend();
+        await fetchRecommendedFriends();
+    } catch (error) {
+        console.error('Error removing friend:', error)
+    }
+};
+
+// search
+const handleSearch = (event) => {
+    const searchTerm = event.target.value.toLowerCase();
+    setSearchTerm(searchTerm);
+
+    //filter friends, recommeded friends and available friend based on search term
+    const filteredFriends = friends.filter(person =>
+        person.name.toLowerCase().includes(searchTerm) ||
+        person.interest.toLowerCase().includes(searchTerm) ||
+        person.school.toLowerCase().includes(searchTerm) ||
+        person.major.toLowerCase().includes(searchTerm)
+        );
+
+    const filteredRecommendedFriends = recommendedFriends.filter(friend =>
+        friend.name.toLowerCase().includes(searchTerm)  ||
+        friend.interest.toLowerCase().includes(searchTerm) ||
+        friend.school.toLowerCase().includes(searchTerm) ||
+        friend.major.toLowerCase().includes(searchTerm)
+        );
+
+    const filteredAvailableFriend = availableFriend.filter(person =>
+        person.name.toLowerCase().includes(searchTerm) ||
+        person.interest.toLowerCase().includes(searchTerm) ||
+        person.school.toLowerCase().includes(searchTerm) ||
+        person.major.toLowerCase().includes(searchTerm)
+        );
+
+    //update filtered data
+    setFilteredData({
+        friends: filteredFriends,
+        recommendedFriends: filteredRecommendedFriends,
+        availableFriend: filteredAvailableFriend
+    });
+};
+
+
+//rendering
+return (
+    <Container maxwidth="sm">
+        <Grid container spacing={2} alignItems="center" justify="center" style={{ minHeight: '80vh'}}>
+            {/*Search bar */}
+            <Grid item xs={12}>
+                <SearchContainer>
+                    <SearchInput
+                    placeholder="Search Friends"
+                    inputProps={{ 'aria-label': 'search Friends'}}
+                    value={searchTerm}
+                    onChange={handleSearch}
+                    />
+                    <IconButton aria-label="search">
+                        <SearchIcon />
+                    </IconButton>
+                </SearchContainer>
+            </Grid>
+            {/* Friends List */}
+            <Grid item xs={12}>
+                <Typography variant="h5">Friends</Typography>
```

Comment:
just a design note for a later PR, but it might make sense to make a separate tab/page for a user's current friends vs. the available+recommended friends so that way their friend management (removing/searching) is distinct from their friend discovery (adding/searching)